go/plugins/compat_oai/generate_live_test.go

```diff
@@ -111,3 +112,61 @@ func TestGenerator_Stream(t *testing.T) {
 	t.Log("\n=== Full Streaming Response ===")
 	t.Log(strings.Join(chunks, ""))
 }
+
+func TestWithConfig(t *testing.T) {
+	tests := []struct {
+		name     string
+		config   any
+		validate func(*testing.T, *openai.ChatCompletionNewParams)
+	}{
+		{
+			name:   "nil config",
+			config: nil,
+			validate: func(t *testing.T, cfg *openai.ChatCompletionNewParams) {
+				// Temperature and MaxTokens should be nil and not present
+				assert.False(t, cfg.Temperature.Present)
+				assert.False(t, cfg.MaxTokens.Present)
+			},
+		},
```

Comment:
Could we add a case for an invalid config - what would the dev experience be for this?

Reply:
test cases added